src/tcgwars/logic/impl/gen8/AmazingVoltTackle.groovy

```diff
@@ -1642,17 +1641,39 @@ public enum AmazingVoltTackle implements LogicCardInfo {
         move "Wrack Down", {
           text "70 damage."
           energyCost D, C, C
-          attackRequirement {}
           onAttack {
             damage 70
           }
         }
         move "Hazard Claw", {
           text "130 damage. Discard 2 Energy from this Pokémon. Your opponent's Active Pokémon is now Paralyzed and Poisoned."
           energyCost D, C, C, C
-          attackRequirement {}
           onAttack {
             damage 130
+            afterDamage {
+              // TODO: Make a static method to do this
+              if (self.cards.energyCount())
+                if (self.cards.energyCount() <= 2) {
+                  self.cards.filterByType(ENERGY).moveTo my.hand
+                } else {
+                  def targetCount = Math.min self.cards.energyCount(), 2
+                  def finalCount = 0
+                  while (self.cards.energyCount() > 0 && finalCount < targetCount) {
+                    def info = "Select Energy to return to your hand."
+                    def energy = self.cards.filterByType(ENERGY).select(info)
+                    def energyCount = 1
+                    if (energy.energyCount() > 1) {
+                      def choices = 1..energy.energyCount()
+                      def choiceInfo = "How many Energy do you want this card to count as?"
+                      energyCount = choose(choices, choiceInfo)
+                    }
+                    finalCount += energyCount
+                    energy.moveTo my.hand
+                  }
+                }
+            }
```

Comment:
```suggestion
            discardSelfEnergy C, C
```

Reply:
This doesn't allow you to choose if you only want a multi energy card to count as a single energy.

> Q. Can I use Umbreon-GX's "Dark Call GX" attack to discard 2 Double Colorless Energy?
A. Yes, you can select 1 Energy of each Double Colorless Energy to discard both of them. This applies for discarding your own Energy for an attack and for retreating as well. So if you used Hoopa-EX's "Hyperspace Fury" attack or paid its Retreat Cost of 2, discarding 2 Double Colorless Energy would be legal options. (Jan 26, 2017 TPCi Rules Team; Jun 27, 2017 TPCi Rules Team)

While solving this in the energy selection would be ideal, and something I plan to do once I have some time to, I think it's better if the card can be used as close to how it can in the paper game until then.

Reply:
I'll work on fixing the energy counting in general ASAP so that these codeblocks can be replaced with just the discardSelfEnergy code, but for now I'm going to merge this as-is since it works correctly this way, and the set releases tomorrow so I'd like to get these things tested.

Reply:
I will fully admit it should discard instead of going to the hand though 😆

Reply:
Hi @flagrama I was confused with the implementation, it mentioned returning energy to hand :) 

I agree this feature should be built-in to `discardSelfEnergy` method call. I am glad you are on top of things, thanks!